src/Functions/array/arrayIndex.h

```diff
@@ -45,8 +53,13 @@ struct IndexCount
     static bool apply(size_t, ResultType & current) { ++current; return true; }
 };
 
+/**
+ * Impls -- how to perform the search depending on the arguments data types.
+ * They usually provide methods @e vector (for processing ordinary columns), @e vector_const (for processing
+ * ConstColumn s), and @e vectorVector for processing vectors of vectors.
+ */
 
-template <typename T, typename U, typename IndexConv>
+template <class Initial, class Result, class ConcreteAction, bool ResizeRes = true>
```

Comment:
Imo, it could be a bit better to give template parameters more descriptive names:
Initial -> LhsValueType
Result -> RhsValueType
ResizeRes -> ResizeResultColumn